core/commands/add.go

```diff
@@ -90,6 +94,17 @@ remains to be implemented.
 
 		progress, _, _ := req.Option(progressOptionName).Bool()
 		wrap, _, _ := req.Option(wrapOptionName).Bool()
+		hidden, _, _ := req.Option(hiddenOptionName).Bool()
+
+		var ignoreFilePatterns []ignore.GitIgnore
+
+		// Check the $IPFS_PATH
+		if ipfs_path := os.Getenv("$IPFS_PATH"); len(ipfs_path) > 0 {
+			baseFilePattern, err := ignore.CompileIgnoreFile(path.Join(ipfs_path, ".ipfsignore"))
+			if err == nil && baseFilePattern != nil {
+				ignoreFilePatterns = append(ignoreFilePatterns, *baseFilePattern)
+			}
+		}
```

Comment:
Good catch on the `$`.

I agree with the premise of `req.Context().ConfigRoot` or something similar. I was asking myself that while coding this up. It seemed like adding a magic path was sub-optimal given that we already had a private folder for the IPFS instance.